sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -236,114 +400,62 @@ export class BatchingReceiver extends MessageReceiver {
         }
 
         try {
-          const data: ServiceBusMessageImpl = this._getServiceBusMessage(context);
-          if (brokeredMessages.length < maxMessageCount) {
+          const data: ServiceBusMessageImpl = this._createServiceBusMessage(context);
+          if (brokeredMessages.length < args.maxMessageCount) {
             brokeredMessages.push(data);
           }
         } catch (err) {
           const errObj = err instanceof Error ? err : new Error(JSON.stringify(err));
           log.error(
-            "[%s] Receiver '%s' received an error while converting AmqpMessage to ServiceBusMessage:\n%O",
-            this._context.namespace.connectionId,
-            this.name,
+            `${loggingPrefix} Received an error while converting AmqpMessage to ServiceBusMessage:\n%O`,
             errObj
           );
           reject(errObj);
         }
-        if (brokeredMessages.length === maxMessageCount) {
+        if (brokeredMessages.length === args.maxMessageCount) {
           finalAction();
         }
       };
 
-      const onSessionClose: OnAmqpEventAsPromise = async (context: EventContext) => {
+      const onClose: OnAmqpEventAsPromise = async (context: EventContext) => {
+        const type = context.session?.error != null ? "session_closed" : "receiver_closed";
+        const error = context.session?.error || context.receiver?.error;
+
         try {
-          const sessionError = context.session && context.session.error;
-          if (sessionError) {
+          if (error) {
             log.error(
-              "[%s] 'session_close' event occurred for receiver '%s'. The associated error is: %O",
-              this._context.namespace.connectionId,
-              this.name,
-              sessionError
+              `${loggingPrefix} '${type}' event occurred. The associated error is: %O`,
+              error
             );
           }
         } catch (err) {
-          log.error(
-            "[%s] Receiver '%s' error in onSessionClose handler:\n%O",
-            this._context.namespace.connectionId,
-            this.name,
-            translate(err)
-          );
+          log.error(`${loggingPrefix} error thrown in ${type} handler:\n%O`, translate(err));
```

Comment:
This catch block and the statement being logged does not make sense to me.
The try block is around an if check and a logging statement.
What do we expect to catch?

Reply:
I just got tripped up by this as well :smile:

Reply:
:) Yeah, I love how re-reviewing this code provides these moments. I _also_ saw this and thought - "well, we can sweep through and get these in one PR that just says "remove overly paranoid error checking").

Reply:
Removed!